compiler/src/iree/compiler/Codegen/Dialect/Codegen/Utils/Utils.h

```diff
@@ -60,6 +62,11 @@ struct TileSwizzle {
     // Support constructing from any size type.
     template <typename T>
     Dim(Kind kind, T size) : kind(kind), size(size) {}
+
+    bool operator==(const Dim &other) const {
```

Comment:
Thanks for the suggestion! I learned more from google c++ style guide, just in case if others are interested in it: https://google.github.io/styleguide/cppguide.html#Operator_Overloading

> Define operators only on your own types. More precisely, define them in the same headers, .cc files, and namespaces as the types they operate on. That way, the operators are available wherever the type is, minimizing the risk of multiple definitions. If possible, avoid defining operators as templates, because they must satisfy this rule for any possible template arguments. If you define an operator, also define any related operators that make sense, and make sure they are defined consistently.

> Prefer to define non-modifying binary operators as non-member functions. If a binary operator is defined as a class member, implicit conversions will apply to the right-hand argument, but not the left-hand one. It will confuse your users if a + b compiles but b + a doesn't.